### ISSUE TITLE
Give all question types non-empty `label`s for screen readers' benefit

### ISSUE THREAD

**Issue by neel-a-mehta** [seattle-uat/civiform#1456]
Screen readers need to see HTML `<label>`s for all `<input>`s so they can read them out. At present, all our question types have `<label>`s, but (as far as I can tell) they are empty.

![image](https://user-images.githubusercontent.com/77128518/123153448-8628ef80-d41a-11eb-8ad5-a7451261afaa.png)

We should [follow this guide](https://www.w3.org/WAI/tutorials/forms/labels/) for putting non-empty, useful labels next to each input. You can use [Tailwind's `sr-only` class](https://tailwindcss.com/docs/screen-readers) to hide the label from non-screen-reading users.

Consider putting the question's title and/or description in the `<label>` next to the `<input>`.

This applies to all question types that use `<input>` tags, which I think is everything.

## Original tester comments

> Actual results:
At step 6 above, observe the 'date question text updated new version updated' dropdown list is unlabeled
> Expected results:
All controls should hae a proper label associated and screenreader should be able to annouce the label , state, controltype & value as seen applicable

**Comment by neel-a-mehta** [comment]
Just tested out "Program for Question Lifecycle" with a screen reader. Looks like it can handle every question type properly, giving a reasonable voiceover description thanks to our `<label>`s. Great!
